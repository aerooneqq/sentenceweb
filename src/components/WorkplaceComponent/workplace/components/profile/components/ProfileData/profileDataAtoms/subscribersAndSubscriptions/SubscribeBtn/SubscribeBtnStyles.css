.subscribeBtn { 
    width: 80px;
    height: 30px;

    background: #12A949;

    border-style: solid;
    border-width: 1px;
    border-radius: 7px;
    border-color: #12A949;

    font-family: Arial;
    color: white;
}

    .subscribeBtn:focus{ 
        outline: none;
    }

    .subscribeBtn:hover{ 
        background: linear-gradient(-120deg, #28B95D, #12A949)
    }