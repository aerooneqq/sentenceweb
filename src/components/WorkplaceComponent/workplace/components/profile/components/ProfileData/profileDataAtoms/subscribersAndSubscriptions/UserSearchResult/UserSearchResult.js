import React, {Component, lazy} from "react";

//Styles
import "../Subscriber/SubscriberStyles.css"

//Components
const SubscribeBtn = lazy(() => import("../SubscribeBtn/SubscribeBtn"));

export default class UserSearchResult extends Component { 
    constructor(props){ 
        super(props);
    }

    render(){ 
        let date = this.props.searchResult.birthDate;
        date = date.substr(0, 4) + "-" + date.substr(5, 2) + "-" + date.substr(8, 2);

        return (
            <div className = "friendsListContainer">
                <div className = "friendsListUserPhoto">
                </div>

                <div className = "friendsListUserInfo">
                    <div className = "friendsListUserName">
                        {this.props.searchResult.name}
                    </div>
                    <div className = "friendsListUserBirthDate">
                        {date}
                    </div>
                    <div class = "friendsListUserBtns">
                        <SubscribeBtn subscribe = {this.props.subscribe} 
                                      userID = {this.props.searchResult.userID}/>
                    </div>
                </div>
            </div>
        );
    }
} 