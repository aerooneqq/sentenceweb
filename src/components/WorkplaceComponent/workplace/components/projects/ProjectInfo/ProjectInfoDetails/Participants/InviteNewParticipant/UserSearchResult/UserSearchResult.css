.user-search-result-container {
    width: 210px;
    height: min-content;

    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: flex-start;   
    
    border-width: 1px;
    border-color: #f97645;
    border-radius: 5px;
    border-style: solid;

    padding-top: 5px;
    padding-bottom: 5px;
    padding-right: 5px;
}

.user-search-result-text-info {
    display: flex;
    flex-direction: column;
    height: 100%;
}

.user-search-result-photo {
    width: 30px;
    height: 30px;

    border-radius: 30px;

    margin-left: 10px;
}

.user-search-result-text-info {
    margin-left: 5px;
}

.user-search-result-name {
    font-size: 17px;
    color: black;
}

.user-search-result-birth-date {
    font-size: 13px;
    color: #909090;
}

.user-search-result-invite-container {
    width: 35px;
    height: 35px;

    border-radius: 35px;

    margin-left: 20px;

    background:  green;

    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
}

    .user-search-result-invite-container:hover {
        background:  rgb(14, 163, 14);
    }