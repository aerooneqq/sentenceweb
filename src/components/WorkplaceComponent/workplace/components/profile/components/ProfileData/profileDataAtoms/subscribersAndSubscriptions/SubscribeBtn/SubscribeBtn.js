import React, {Component} from "react";

//Styles
import "./SubscribeBtnStyles.css";

export default class SubscribeBtn extends Component { 
    constructor(props) { 
        super(props);

        this.handleBtnClick = this.handleBtnClick.bind(this);
    }

    handleBtnClick(){ 
        this.props.subscribe(this.props.userID);
    }

    render(){ 
        return ( 
            <button className = "subscribeBtn"
                    onClick = {this.handleBtnClick}>
                Subscribe
            </button>
        )
    }
}