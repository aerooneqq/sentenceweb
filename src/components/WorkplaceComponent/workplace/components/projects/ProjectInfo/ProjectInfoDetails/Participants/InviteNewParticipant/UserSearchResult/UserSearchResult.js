import React, {Component} from "react";

import "./UserSearchResult.css";

import inviteUserIcon from "./img/invite_user.svg";

import UserPhotoService from "../../../../../../../../../../services/UserPhotoService/UserPhotoService";
import ResponseService from "../../../../../../../../../../services/ResponseService/ReponseService";


export default class UserSearchResult extends Component {
    constructor(props) {
        super(props);

        this.state = {
            userPhoto: null
        }

        this.userPhotoService = new UserPhotoService(localStorage.getItem("token"));
        this.responseService = new ResponseService();
    }

    componentDidMount() {
        this.userPhotoService.getOtherUserPhoto(this.props.user.userID)
            .then(res => {
                this.setState({
                    userPhoto: res.data,
                })
            })
            .catch(err => {
                this.responseService.alertErrorMessage(err, "Error ocurred while getting user photo");
            })
    }

    render() {
        return (
            <div className = "user-search-result-container">
                <img className = "user-search-result-photo" src = {"data:image/png;base64," + this.state.userPhoto} />
                <div className = "user-search-result-text-info">
                    <div className = "user-search-result-name">
                        {this.props.user.name}
                    </div>
                    <div className = "user-search-result-birth-date">
                        {this.props.user.birthDate.substr(0, 10)}
                    </div>
                </div>
                <div className = "user-search-result-invite-container">
                    <img className = "user-search-result-invite-icon" src = {inviteUserIcon} />
                </div>
            </div>
        )
    }
}