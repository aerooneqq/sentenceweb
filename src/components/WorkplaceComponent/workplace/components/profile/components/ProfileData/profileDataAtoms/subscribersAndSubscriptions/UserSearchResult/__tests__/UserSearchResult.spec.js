import React from "react";
import {shallow} from "enzyme";
import {shallowToJson} from "enzyme-to-json";
import UserSearchResult from "../UserSearchResult";

import SubscribeBtn from "../../SubscribeBtn/SubscribeBtn";

const props = { 
    searchResult: { 
        name: "name",
        birthDate: "30.05.2000T21312312312312",
        userID: 1,
    },
    subscribe: () => {}
}

describe("UserSearchResult", () => { 
    it("should render correctly", () => { 
        expect(shallowToJson(shallow(<UserSearchResult {...props}/>))).toMatchSnapshot();
    });

    it("should display right info about user", () => { 
        const userSearchResult = shallow(<UserSearchResult {...props}/>);

        expect(userSearchResult.find("div.friendsListUserName").text()).toBe(props.searchResult.name);
        
        let {searchResult: {birthDate}} = props;
        birthDate = birthDate.substr(0, 4) + "-" + birthDate.substr(5, 2) + "-" + birthDate.substr(8, 2);
        
        expect(userSearchResult.find("div.friendsListUserBirthDate").text()).toBe(birthDate);
    });  

    it("should have subscribe btn", () => { 
        const userSearchResult = shallow(<UserSearchResult {...props} />);

        expect(userSearchResult.find(SubscribeBtn).exists()).toBe(true);
    })
});