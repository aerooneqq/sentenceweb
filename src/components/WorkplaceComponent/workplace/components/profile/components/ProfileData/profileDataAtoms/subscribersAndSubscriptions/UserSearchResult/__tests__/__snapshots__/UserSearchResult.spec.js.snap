// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSearchResult should render correctly 1`] = `
<div
  className="friendsListContainer"
>
  <div
    className="friendsListUserPhoto"
  />
  <div
    className="friendsListUserInfo"
  >
    <div
      className="friendsListUserName"
    >
      name
    </div>
    <div
      className="friendsListUserBirthDate"
    >
      30.0-.2-00
    </div>
    <div
      class="friendsListUserBtns"
    >
      <SubscribeBtn
        subscribe={[Function]}
        userID={1}
      />
    </div>
  </div>
</div>
`;
